### DISCUSSION TITLE
Need help installing box64 on Centos7 on Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by mdeal5** [ptitSeb/box64#944] [Q&A] [answered]
Folks, I'm an admitted newbie and by all appearances am in over my head.
Happy to rtfm, but I can't find anything to read that provides any assistance or direction.

When I'm trying to run the make box 64  using make -j$(nproc)  I get the following errors.
cc: error: unrecognized argument in option '-march=armv8-a+crc'
and
cc: error: unrecognized argument in option '-mtune=cortex-a72'

Through the research I've done, my Raspberry Pi 4 is running ARMv7 rev 3

What I can't find is information on where to correct the parms in error in the makefiles or the config files so that I can get this thing to complete.

Any advice at all would be greatly appreciated.

Thanks in advance.

**Comment by ptitSeb** [comment]
If you are on 32bits OS, there isn't much you can do. box64 is exclusively for 64bits OS.

**Reply by mdeal5** [reply]
I'm on a 64 bit version of Centos 7.

**Reply by mdeal5** [reply]
Specifically on CentOS-Userland-7-armv7hl-RaspberryPI-Minimal-4-2003-sda.raw.xz.  It is 64 bit.

**Reply by mdeal5** [reply]
Also, it only gives the failure message on 2 files, but I didn't write that information down and will have to go through the install again to get them.
My confusion being that after running the command
cmake .. -DRPI4ARM64=1 -DCMAKE_BUILD_TYPE=RelWIthDebInfo
It is supplying the parameters for armv8-a+crc instead of just armv8-a, which is an acceptable variable.
And I can't find where to go to change that or even view the entry.
Similar issue on the -mtune argument.

Regardless, thank you for the input, but I'm not quite THAT much of a noob. :-)

**Reply by mdeal5** [reply]
Last comment.  I re-ran the make -j$(nproc) command.  It is successful on dynablock.c.o
and fails on 
CMakeFiles/test_interpretir.dir/src/emu/x64run.c.o
multiple times.
I'm not sufficiently familiar with the product to give any further information, but am happy to provide a better description.
Here's the output
[mark@unknowne45f012e507a build]$ make -j$(nproc)
[  1%] Building C object CMakeFiles/dynarec_native.dir/src/dynarec/dynablock.c.o
[  1%] Building C object CMakeFiles/test_interpreter.dir/src/emu/x64run.c.o
cc: error: unrecognized argument in option ‘-march=armv8-a+crc’
cc: note: valid arguments to ‘-march=’ are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5e armv5t armv5te armv6 armv6-m armv6j armv6k armv6s-m armv6t2 armv6z armv6zk armv7 armv7-a armv7-m armv7-r armv7e-m armv8-a iwmmxt iwmmxt2 native
cc: error: unrecognized argument in option ‘-mtune=cortex-a72’
cc: note: valid arguments to ‘-mtune=’ are: arm1020e arm1020t arm1022e arm1026ej-s arm10e arm10tdmi arm1136j-s arm1136jf-s arm1156t2-s arm1156t2f-s arm1176jz-s arm1176jzf-s arm2 arm250 arm3 arm6 arm60 arm600 arm610 arm620 arm7 arm70 arm700 arm700i arm710 arm7100 arm710c arm710t arm720 arm720t arm740t arm7500 arm7500fe arm7d arm7di arm7dm arm7dmi arm7m arm7tdmi arm7tdmi-s arm8 arm810 arm9 arm920 arm920t arm922t arm926ej-s arm940t arm946e-s arm966e-s arm968e-s arm9e arm9tdmi cortex-a15 cortex-a5 cortex-a7 cortex-a8 cortex-a9 cortex-m0 cortex-m0plus cortex-m1 cortex-m3 cortex-m4 cortex-r4 cortex-r4f cortex-r5 ep9312 fa526 fa606te fa626 fa626te fa726te fmp626 generic-armv7-a iwmmxt iwmmxt2 marvell-pj4 mpcore mpcorenovfp native strongarm strongarm110 strongarm1100 strongarm1110 xscale
make[2]: *** [CMakeFiles/dynarec_native.dir/src/dynarec/dynablock.c.o] Error 1
make[1]: [  1%] Building C object CMakeFiles/interpreter.dir/src/emu/x64run.c.o
*** [CMakeFiles/dynarec_native.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
cc: error: unrecognized argument in option ‘-march=armv8-a+crc’
cc: note: valid arguments to ‘-march=’ are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5e armv5t armv5te armv6 armv6-m armv6j armv6k armv6s-m armv6t2 armv6z armv6zk armv7 armv7-a armv7-m armv7-r armv7e-m armv8-a iwmmxt iwmmxt2 native
cc: error: unrecognized argument in option ‘-mtune=cortex-a72’
cc: note: valid arguments to ‘-mtune=’ are: arm1020e arm1020t arm1022e arm1026ej-s arm10e arm10tdmi arm1136j-s arm1136jf-s arm1156t2-s arm1156t2f-s arm1176jz-s arm1176jzf-s arm2 arm250 arm3 arm6 arm60 arm600 arm610 arm620 arm7 arm70 arm700 arm700i arm710 arm7100 arm710c arm710t arm720 arm720t arm740t arm7500 arm7500fe arm7d arm7di arm7dm arm7dmi arm7m arm7tdmi arm7tdmi-s arm8 arm810 arm9 arm920 arm920t arm922t arm926ej-s arm940t arm946e-s arm966e-s arm968e-s arm9e arm9tdmi cortex-a15 cortex-a5 cortex-a7 cortex-a8 cortex-a9 cortex-m0 cortex-m0plus cortex-m1 cortex-m3 cortex-m4 cortex-r4 cortex-r4f cortex-r5 ep9312 fa526 fa606te fa626 fa626te fa726te fmp626 generic-armv7-a iwmmxt iwmmxt2 marvell-pj4 mpcore mpcorenovfp native strongarm strongarm110 strongarm1100 strongarm1110 xscale
make[2]: *** [CMakeFiles/test_interpreter.dir/src/emu/x64run.c.o] Error 1
make[1]: *** [CMakeFiles/test_interpreter.dir/all] Error 2
cc: error: unrecognized argument in option ‘-march=armv8-a+crc’
cc: note: valid arguments to ‘-march=’ are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5e armv5t armv5te armv6 armv6-m armv6j armv6k armv6s-m armv6t2 armv6z armv6zk armv7 armv7-a armv7-m armv7-r armv7e-m armv8-a iwmmxt iwmmxt2 native
cc: error: unrecognized argument in option ‘-mtune=cortex-a72’
cc: note: valid arguments to ‘-mtune=’ are: arm1020e arm1020t arm1022e arm1026ej-s arm10e arm10tdmi arm1136j-s arm1136jf-s arm1156t2-s arm1156t2f-s arm1176jz-s arm1176jzf-s arm2 arm250 arm3 arm6 arm60 arm600 arm610 arm620 arm7 arm70 arm700 arm700i arm710 arm7100 arm710c arm710t arm720 arm720t arm740t arm7500 arm7500fe arm7d arm7di arm7dm arm7dmi arm7m arm7tdmi arm7tdmi-s arm8 arm810 arm9 arm920 arm920t arm922t arm926ej-s arm940t arm946e-s arm966e-s arm968e-s arm9e arm9tdmi cortex-a15 cortex-a5 cortex-a7 cortex-a8 cortex-a9 cortex-m0 cortex-m0plus cortex-m1 cortex-m3 cortex-m4 cortex-r4 cortex-r4f cortex-r5 ep9312 fa526 fa606te fa626 fa626te fa726te fmp626 generic-armv7-a iwmmxt iwmmxt2 marvell-pj4 mpcore mpcorenovfp native strongarm strongarm110 strongarm1100 strongarm1110 xscale
make[2]: *** [CMakeFiles/interpreter.dir/src/emu/x64run.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[  1%] Building C object CMakeFiles/interpreter.dir/src/emu/x64run0f.c.o
[  1%] Built target WRAPPERS
cc: error: unrecognized argument in option ‘-march=armv8-a+crc’
cc: note: valid arguments to ‘-march=’ are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5e armv5t armv5te armv6 armv6-m armv6j armv6k armv6s-m armv6t2 armv6z armv6zk armv7 armv7-a armv7-m armv7-r armv7e-m armv8-a iwmmxt iwmmxt2 native
cc: error: unrecognized argument in option ‘-mtune=cortex-a72’
cc: note: valid arguments to ‘-mtune=’ are: arm1020e arm1020t arm1022e arm1026ej-s arm10e arm10tdmi arm1136j-s arm1136jf-s arm1156t2-s arm1156t2f-s arm1176jz-s arm1176jzf-s arm2 arm250 arm3 arm6 arm60 arm600 arm610 arm620 arm7 arm70 arm700 arm700i arm710 arm7100 arm710c arm710t arm720 arm720t arm740t arm7500 arm7500fe arm7d arm7di arm7dm arm7dmi arm7m arm7tdmi arm7tdmi-s arm8 arm810 arm9 arm920 arm920t arm922t arm926ej-s arm940t arm946e-s arm966e-s arm968e-s arm9e arm9tdmi cortex-a15 cortex-a5 cortex-a7 cortex-a8 cortex-a9 cortex-m0 cortex-m0plus cortex-m1 cortex-m3 cortex-m4 cortex-r4 cortex-r4f cortex-r5 ep9312 fa526 fa606te fa626 fa626te fa726te fmp626 generic-armv7-a iwmmxt iwmmxt2 marvell-pj4 mpcore mpcorenovfp native strongarm strongarm110 strongarm1100 strongarm1110 xscale
make[2]: *** [CMakeFiles/interpreter.dir/src/emu/x64run0f.c.o] Error 1
[  1%] Building C object CMakeFiles/interpreter.dir/src/emu/x64run64.c.o
cc: error: unrecognized argument in option ‘-march=armv8-a+crc’
cc: note: valid arguments to ‘-march=’ are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5e armv5t armv5te armv6 armv6-m armv6j armv6k armv6s-m armv6t2 armv6z armv6zk armv7 armv7-a armv7-m armv7-r armv7e-m armv8-a iwmmxt iwmmxt2 native
cc: error: unrecognized argument in option ‘-mtune=cortex-a72’
cc: note: valid arguments to ‘-mtune=’ are: arm1020e arm1020t arm1022e arm1026ej-s arm10e arm10tdmi arm1136j-s arm1136jf-s arm1156t2-s arm1156t2f-s arm1176jz-s arm1176jzf-s arm2 arm250 arm3 arm6 arm60 arm600 arm610 arm620 arm7 arm70 arm700 arm700i arm710 arm7100 arm710c arm710t arm720 arm720t arm740t arm7500 arm7500fe arm7d arm7di arm7dm arm7dmi arm7m arm7tdmi arm7tdmi-s arm8 arm810 arm9 arm920 arm920t arm922t arm926ej-s arm940t arm946e-s arm966e-s arm968e-s arm9e arm9tdmi cortex-a15 cortex-a5 cortex-a7 cortex-a8 cortex-a9 cortex-m0 cortex-m0plus cortex-m1 cortex-m3 cortex-m4 cortex-r4 cortex-r4f cortex-r5 ep9312 fa526 fa606te fa626 fa626te fa726te fmp626 generic-armv7-a iwmmxt iwmmxt2 marvell-pj4 mpcore mpcorenovfp native strongarm strongarm110 strongarm1100 strongarm1110 xscale
make[2]: *** [CMakeFiles/interpreter.dir/src/emu/x64run64.c.o] Error 1
make[1]: *** [CMakeFiles/interpreter.dir/all] Error 2
make: *** [all] Error 2

**Comment by mdeal5** [comment, marked as answer]
Belay everything that I posted previously.  Through some additional digging I found the CMakeLists.txt file and got past that hurdle.
Now on to the next.
